relay-dynamic-config/src/trusted_relay.rs

```diff
@@ -0,0 +1,84 @@
+use serde::{Deserialize, Serialize};
+
+/// Configuration to control communication from trusted relays.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(default, rename_all = "camelCase")]
+pub struct TrustedRelayConfig {
+    /// Checks the signature of an event and rejects it if enabled.
+    #[serde(skip_serializing_if = "SignatureVerification::is_default")]
+    pub verify_signature: SignatureVerification,
+}
+
+impl TrustedRelayConfig {
+    pub fn is_empty(&self) -> bool {
+        self.verify_signature == SignatureVerification::default()
+    }
+}
+
+/// Types of verification that can be performed on the signature.
+#[derive(Debug, Clone, Default, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum SignatureVerification {
+    /// Checks the signature for validity and verifies that the embedded timestamp
+    /// is not too old.
+    WithTimestamp,
+    /// Does not perform any validation on the signature.
+    #[default]
+    Disabled,
+}
+
+impl SignatureVerification {
+    pub fn is_default(&self) -> bool {
+        *self == SignatureVerification::default()
+    }
+
+    pub fn is_verify_timestamp(&self) -> bool {
+        *self == SignatureVerification::WithTimestamp
+    }
+
+    pub fn is_disabled(&self) -> bool {
+        *self == SignatureVerification::Disabled
+    }
```

Comment:
Missing docstrings, but maybe it's just better to let the caller do the comparisons (except is default), the function really doesn't do much.